### PR TITLE
CAT-2327 Enables formula editor in setLookBrick

### PR DESCRIPTION
**Pull request by comeclarity**
The dropdown menu from the setLookBrick now has an option to open the formula editor to choose a look by formula (look_number or look_name)

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/SetLookBrick.java [hunk 1]
```diff
@@ -115,12 +123,13 @@ public View getView(final Context context, int brickId, BaseAdapter baseAdapter)
...
 			@Override
 			public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
-				if (position == 0) {
+				if (position == NEW_LOOK_POSITION) {
 					look = null;
 				} else {
 					look = (LookData) parent.getItemAtPosition(position);
 					oldSelectedLook = look;
 				}
+				isFormulaEnabled = (position == FORMULA_EDITOR_POSITION);
```
Comment (cheidenreich) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/SetLookBrick.java]: maybe change the naming of the variables to `SPINNER_POSITION_NEW_LOOK` and `SPINNER_POSITION_FORMULA_EDITOR` to make it clear which position are meant

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/SetLookBrick.java [hunk 2]
```diff
@@ -175,31 +195,47 @@ public View getPrototypeView(Context context) {
...
 		if (getSprite().getLookDataList().contains(look)) {
 			oldSelectedLook = look;
-			spinner.setSelection(getSprite().getLookDataList().indexOf(look) + 1, true);
+			spinner.setSelection(getSprite().getLookDataList().indexOf(look) + 2, true);
 		} else {
 			if (spinner.getAdapter() != null && spinner.getAdapter().getCount() > 1) {
 				if (getSprite().getLookDataList().indexOf(oldSelectedLook) >= 0) {
 					spinner.setSelection(getSprite().getLookDataList()
-							.indexOf(oldSelectedLook) + 1, true);
+							.indexOf(oldSelectedLook) + 2, true);
```
Comment (cheidenreich) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/SetLookBrick.java]: You can also use `FORMULA_EDITOR_POSITION + 1` here

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/SetLookBrick.java [hunk 3]
```diff
@@ -262,10 +298,15 @@ public boolean hasStableIds() {
 
 		@Override
 		public View getView(int paramInt, View paramView, ViewGroup paramViewGroup) {
+			spinnerAdapter.getItem(FORMULA_EDITOR_POSITION).setLookName(getFormulaWithBrickField(BrickField
+					.LOOK).getDisplayString(context));
 			if (isTouchInDropDownView) {
 				isTouchInDropDownView = false;
 				if (paramInt == 0) {
 					switchToLookFragmentFromScriptFragment();
+				} else if (paramInt == 1) {
```
Comment (cheidenreich) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/SetLookBrick.java]: `FORMULA_EDITOR_POSITION` can be used here

**Comment by comeclarity** [comment]
Thanks for the suggestions. I updated the pull request :)

File: catroid/src/androidTest/java/org/catrobat/catroid/test/content/actions/SetLookActionTest.java [hunk 4]
```diff
@@ -100,4 +101,33 @@ public void testSetLook() {
...
+
+		Formula numberFormula = new Formula(2);
+
+		ActionFactory factory = sprite.getActionFactory();
+		Action action = factory.createSetLookAction(sprite, firstLookData, false, numberFormula);
+		action.act(1.0f);
+		assertEquals("Action didn't set the LookData correctly from the number formula", secondLookData, sprite.look
+				.getLookData());
+
+		Formula stringFormula = new Formula("firstTestLook");
```
Comment (rschluesselbauer) [inline catroid/src/androidTest/java/org/catrobat/catroid/test/content/actions/SetLookActionTest.java]: According to the article series on testable code this part should probably be put in a seperate test, as there should not be acting in between asserts (anatomy of a testcase: assemble - act - assert).

Comment (comeclarity) [inline catroid/src/androidTest/java/org/catrobat/catroid/test/content/actions/SetLookActionTest.java]: Good point, thank you.
I updated the tests.

File: catroid/src/main/java/org/catrobat/catroid/content/bricks/SetLookBrick.java [hunk 5]
```diff
@@ -175,31 +194,47 @@ public View getPrototypeView(Context context) {
...
+			spinner.setSelection(getSprite().getLookDataList().indexOf(look) + 2, true);
 		} else {
 			if (spinner.getAdapter() != null && spinner.getAdapter().getCount() > 1) {
 				if (getSprite().getLookDataList().indexOf(oldSelectedLook) >= 0) {
 					spinner.setSelection(getSprite().getLookDataList()
-							.indexOf(oldSelectedLook) + 1, true);
+							.indexOf(oldSelectedLook) + SPINNER_POSITION_FORMULA_EDITOR + 1, true);
 				} else {
-					spinner.setSelection(1, true);
+					spinner.setSelection(SPINNER_POSITION_FORMULA_EDITOR + 1, true);
```
Comment (rschluesselbauer) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/SetLookBrick.java]: App crashes here for me.
To reproduce, create a new empty project (there should not be any background images!) and try to add a setlookbrick. When the brick is displayed in the menu, the application crashes for me.

Comment (comeclarity) [inline catroid/src/main/java/org/catrobat/catroid/content/bricks/SetLookBrick.java]: Fixed it. (was due to an off-by-one error in the if-condition)
Thank you

**Comment by bernadettespieler** [comment]
new UX discussions...

**Pull request closed without merging**